posts/hello-microservice-deployment-part-2.md

```diff
@@ -0,0 +1,358 @@
+
+This was a very very brief introduction to K8s objects and their capabilities. Again there are [entire books](https://www.amazon.com/gp/product/1491935677/ref=as_li_tl?ie=UTF8&camp=1789&creative=9325&creativeASIN=1491935677&linkCode=as2&tag=sheena0d-20&linkId=e5bf88f134212c4beba73c601598d7ff") written on the subject and if you want to become an expert on the topic I would urge you to do some reading on your own.
+
+## Practical: Deploying our application to Google cloud
+
+Now we know why a tool like K8s is useful... let's start using it. I've chosen to proceed by making use of Google's Kubernetes engine offering. You can use a different service provider if you want to, or even run your own cluster, or simulate a cluster using [minikube](https://kubernetes.io/docs/setup/minikube/). For the most part the tutorial will be exactly the same except for the initial cluster setup and the final cluster cleanup.
+
+### Set up our cluster
+
+Visit the Kubernetes engine page (https://console.cloud.google.com/projectselector/kubernetes). If you are new to Google cloud you'll need to sign in with your Google account. You can sign up for $300 worth of free credits while you are at it. If you aren't new to Google and have already used up your free credits then the cluster will cost money (not a lot though).
+
+Create or select a project - their user interface is fairly intuitive. You'll need to wait a little while for the K8s API and related services to be activated.
+
+Now visit the [Google Cloud Console](https://console.cloud.google.com). At the top of the page you will see a little terminal button. Click on it to activate the Google cloud shell.
+
+If you want to use a terminal on your own computer then you can, it just requires a bit more setup. You would need to install the Google Cloud SDK and the kubectl component and perform some extra configuration that's outside the scope of this text. If you want to go down that route take a look at [this](https://cloud.google.com/sdk/docs/quickstarts). You'll be interacting with your cluster through use of the gcloud command line utility.
+
+Ok, so back to the shell: Run the following command to create your cluster:
+```
+gcloud container clusters create hello-timber --num-nodes=3 --zone=europe-west1-d
+```
+
+This will create a three node cluster called hello-timber. It might take a few minutes. Each node is a compute instance (VM) managed by Google. I've chosen here to put the nodes in west Europe. If you want to see a full list of available zones you can use the command `gcloud compute zones list`. Alternatively you can refer to [this document](https://cloud.google.com/compute/docs/regions-zones/) for a list of available zones.
+
+In general it is good practice to choose a zone that is close to whatever clients are consuming your application. Although it is possible to deploy an application on multiple clusters in multiple zones around the world, that is waaaay outside the scope of this tutorial.
```

Comment:
This isn't necessary.

Reply:
deleted.

I figure it is nice to allude to topics that are outside of scope, but possibly interesting to users. It sparks curiosity and, if readers show interest, articles can be written on the subject